fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadJob.java

```diff
@@ -0,0 +1,700 @@
+        switch (status.getState()) {
+            case RUNNING:
+                updateEtlStatusInternal(status);
+                break;
+            case FINISHED:
+                processEtlFinish(status, handler);
+                break;
+            case CANCELLED:
+                throw new LoadException("spark etl job failed. msg: " + status.getFailMsg());
+            default:
+                LOG.warn("unknown etl state: {}", status.getState().name());
+                break;
+        }
+    }
+
+    private void updateEtlStatusInternal(EtlStatus etlStatus) {
+        writeLock();
+        try {
+            loadingStatus = etlStatus;
+            progress = etlStatus.getProgress();
+            if (!sparkResource.isYarnMaster()) {
+                loadingStatus.setTrackingUrl(appId);
+            }
+
+            // TODO(wyb): spark-load
+            /*
+            DppResult dppResult = etlStatus.getDppResult();
+            if (dppResult != null) {
+                // update load statistic and counters when spark etl job finished
+                // fe gets these infos from spark dpp, so we use dummy load id and dummy backend id here
+                loadStatistic.fileNum = (int) dppResult.fileNumber;
+                loadStatistic.totalFileSizeB = dppResult.fileSize;
+                TUniqueId dummyId = new TUniqueId(0, 0);
+                long dummyBackendId = -1L;
+                loadStatistic.initLoad(dummyId, Sets.newHashSet(dummyId), Lists.newArrayList(dummyBackendId));
+                loadStatistic.updateLoadProgress(dummyBackendId, dummyId, dummyId, dppResult.scannedRows, true);
+
+                Map<String, String> counters = loadingStatus.getCounters();
+                counters.put(DPP_NORMAL_ALL, String.valueOf(dppResult.normalRows));
+                counters.put(DPP_ABNORMAL_ALL, String.valueOf(dppResult.abnormalRows));
+                counters.put(UNSELECTED_ROWS, String.valueOf(dppResult.unselectRows));
+            }
+            */
+        } finally {
+            writeUnlock();
+        }
+    }
+
+    private void processEtlFinish(EtlStatus etlStatus, SparkEtlJobHandler handler) throws Exception {
+        updateEtlStatusInternal(etlStatus);
+        // checkDataQuality
+        if (!checkDataQuality()) {
+            cancelJobWithoutCheck(new FailMsg(FailMsg.CancelType.ETL_QUALITY_UNSATISFIED, QUALITY_FAIL_MSG),
+                                  true, true);
+            return;
+        }
+
+        // get etl output files and update loading state
+        updateToLoadingState(etlStatus, handler.getEtlFilePaths(etlOutputPath, brokerDesc));
+        // log loading state
+        logUpdateStateInfo();
+
+        // create push tasks
+        prepareLoadingInfos();
+        submitPushTasks();
+    }
+
+    private void updateToLoadingState(EtlStatus etlStatus, Map<String, Long> filePathToSize) throws LoadException {
+        writeLock();
```

Comment:
why not get lock in the outer public function? In most scenario, there is only one writer.
If you put this lock in a private function, there is a case that multiple threads will do proceeEtlFinish concurrently.